site/content/docs/03-run-time.md

```diff
@@ -50,6 +50,8 @@ If a function is returned from `onMount`, it will be called when the component i
 </script>
 ```
 
+>  Returning a function is only available for *synchronous* onMount functions. If an `async` function or `Promise` is passed to `onMount`, the returned function will **not** be called.
```

Comment:
I do think we ought to clarify to people here how this works, but I'm not sold on this particular wording. Maybe instead just saying something about how 'the function passed to `onMount` needs to _synchronously_ return a function for it to be called when the component is destroyed'? And then maybe a reminder that `async` functions always actually return promises.

Reply:
Updated. It's very hard to get the wording right in a concise way without requiring the user to have a very solid grounding in asynchronous functions, anonymous functions, and Promise execution order.

I think I've covered it though.